flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -333,6 +337,17 @@ private void performAuthorization(AuthorizationServiceConfiguration serviceConfi
                 authRequestBuilder.setUiLocales(additionalParameters.get("ui_locales"));
                 additionalParameters.remove("ui_locales");
             }
+
+            if(additionalParameters.containsKey("claims")){
+                try {
+                    final JSONObject claimsAsJson = new JSONObject(additionalParameters.get("claims"));
+                    authRequestBuilder.setClaims(claimsAsJson);
+                    additionalParameters.remove("claims");
+                }
+                catch (JSONException ex) {
+                    finishWithError(INVALID_CLAIMS_ERROR_CODE, ex.getLocalizedMessage(), getCauseFromException(ex));
```

Comment:
Given how the code is written, if an exception occurs, it'll be caught and dealt with but will continue to remaining logic in this method. You'd need to add a `return` statement to prevent this from happening. The `finishWithError()` call is more to notify the Flutter/Dart about the progress of the method

Reply:
Good catch - will make it so.